Table generation with an extra blank line

[edit: code blocks for readability]

The following doxygen c++ comment

```cpp
/// | A | B | A && B |
/// |---|---|--------|
/// | F | F |   F    |
/// | F | T |   F    |
/// | T | F |   F    |
/// | T | T |   T    |
```
generates the following xml
```xml
<para><table rows="5" cols="3"><row>
<entry thead="yes"><para>A  </para>
</entry><entry thead="yes"><para>B  </para>
</entry><entry thead="yes"><para>A &amp;&amp; B   </para>
</entry></row>
<row>
<entry thead="no"><para>F  </para>
</entry><entry thead="no"><para>F  </para>
</entry><entry thead="no"><para>F   </para>
</entry></row>
<row>
...
```

doxyrest produces

```rst
=  =  ====== 
A  B  A && B 
=  =  ====== 

F  F  F      
F  T  F      
T  F  F      
T  T  T      
=  =  ======
```

There is an extra blank line after the table header line which breaks the rendering.

I'll be happy to fix it if you think this doesn't require complex surgery.

Hello @vovkos ,

I am having some difficulty building doxyrest from the source. I am facing the following problems:
- doxyrest_b repo
CMake Error at axl/cmake/axl_utils.cmake:1318 (message):
  ragel is REQUIRED but NOT FOUND, adjust paths.cmake
Call Stack (most recent call first):
  axl/cmake/axl_utils.cmake:1342 (axl_include_import_file)
  axl/cmake/axl_init.cmake:99 (axl_import)
  axl/cmake/axl_cmake.cmake:131 (include)
  CMakeLists.txt:16 (include)

- doxyrest
CMake Error at /home/ghochee/bin/src/axl/cmake/axl_utils.cmake:1318 (message):
  axl is REQUIRED but NOT FOUND, adjust AXL_CMAKE_DIR in paths.cmake
Call Stack (most recent call first):
  <path-to-axl-clone>/cmake/axl_utils.cmake:1342 (axl_include_import_file)
  <path-to-axl-clone>/cmake/axl_init.cmake:99 (axl_import)
  cmake/axl_cmake.cmake:131 (include)
  CMakeLists.txt:16 (include)

I built and installed expat library on a systemwide path
I built and installed axl library and headers on a systemwide path
ls -l libaxl*
-rw-r--r-- 1 root root 864058 Jul  7 11:43 libaxl_core.a
-rw-r--r-- 1 root root  94320 Jul  7 11:43 libaxl_cry.a
-rw-r--r-- 1 root root 178338 Jul  7 11:43 libaxl_fsm.a
-rw-r--r-- 1 root root 161982 Jul  7 11:43 libaxl_gui.a
-rw-r--r-- 1 root root 481740 Jul  7 11:43 libaxl_io.a
-rw-r--r-- 1 root root  19038 Jul  7 11:43 libaxl_lex.a
-rw-r--r-- 1 root root  29612 Jul  7 11:43 libaxl_lua.a
-rw-r--r-- 1 root root  39710 Jul  7 11:43 libaxl_xml.a
-rw-r--r-- 1 root root 114960 Jul  7 11:43 libaxl_zip.a
Note there is no libaxl.* . AFAICT this is what is required by the doxyrest CMakeLists.txt file.

I think maybe the projects are too out of sync so thought of reaching out before attempting other more complicated things. Thanks in advance for your help on this.

- Aditya

--- a/frame/common/doc.lua
+++ b/frame/common/doc.lua
@@ -384,18 +384,20 @@ function formatDocBlock_table(b, context)
 	for i = 1, #maxwidths do
 		headfoot = headfoot .. string.rep('=', maxwidths[i]) .. '  '
 	end
+	headfoot = headfoot .. '\n'
 
-	local s = headfoot .. '\n'
-
+	local s = headfoot
 	for r = 1, #tbl do
 		for c = 1, #tbl[r] do
 			s = s .. tbl[r][c] .. string.rep(' ', 2 + maxwidths[c] - string.len(tbl[r][c]))
 		end
 		s = s .. '\n'
-		if r == 1 or r == #tbl then
-			s = s .. headfoot .. '\n\n'
+
+		if r == 1 then
+			s = s .. headfoot
 		end
 	end
+	s = s .. headfoot
 
 	return s
 end

